travis-ci.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+set +e
+
+if [ -z "$TRAVIS" ]; then
+    echo "this script is intended to be run only on travis" >&2
+    exit 1
+fi
+
+KUBECTL_WAIT_CMD="kubectl wait --timeout=5m --for=condition=Ready pod/sonobuoy -n heptio-sonobuoy"
+SONOBUOY_CMD=./sonobuoy
+SONOBUOY_SUB_CMD=run -m Quick
+
+function waitForDeployment() {
+  echo "|---- Waiting for deployment to be 'Ready'..."
+
+  # Attempt get sonobuoy pod for readiness
+  $KUBECTL_WAIT_CMD
+  while [ $? -eq 0 ] 
+  do
+    if [ $? -eq 0 ]
+    then
+      break
+    else
+      echo "sonobuoy not ready..."
+      sleep 2
+      # Attempt get sonobuoy pod for readiness
+      $KUBECTL_WAIT_CMD
+    fi
+  done
+}
+
+function checkForResults() {
+  echo "|---- Looking for results..."
+
+  # Attempt to get status
+  $SONOBUOY_CMD retrieve
+  if [ -n "$(ls -A *.gz 2>/dev/null)" ]
+  then
+    echo "found results!"
+    return
+  fi
+
+  echo "results not ready..."
+  false
+}
+
+echo "|---- Creating new Sonobuoy run..."
+$SONOBUOY_CMD run
+waitForDeployment
+
+while [ !checkForResults ]
+do
+  $SONOBUOY_CMD retrieve
+  $SONOBUOY_CMD status
+  sleep 30
+done
+
+echo "|------ Results found, validating output..."
+
+for i in *.gz; do
+  mkdir -p results
+  tar -xvf $i -C results
+  grep -q "Test Suite Passed" results/plugins/e2e/results/e2e.log
+  
+  if [ $? -eq 0 ]
+  then
+    echo "Success!"
+    exit 0
+  else
+    echo "Failure: E2E tests failed" >&2
```

Comment:
Not sure the file retention via jenkins here, but if we can't grab the tar then we need to be sure to tail/echo the end of the file where all the failed tests are listed otherwise we'll be perpetually frustrated by not knowing which tests failed.